zebra-state/src/service.rs

```diff
@@ -803,6 +803,91 @@ impl ReadStateService {
     fn latest_non_finalized_state(&self) -> NonFinalizedState {
         self.non_finalized_state_receiver.cloned_watch_data()
     }
+
+    /// Check the contextual validity of a block in the best chain
```

Comment:
Which consensus rules are checked here? Do we skip any?

Reply:
All of the usual ones that are checked in zebra-state for committing a block to the non-finalized state. 

> Do we skip any?

Certainly not now that it's using `write::validate_and_commit_non_finalized`.